rust/private/rust.bzl

```diff
@@ -384,20 +384,20 @@ def _rust_binary_impl(ctx):
         ),
     )
 
-def _rust_test_common(ctx, toolchain, output):
+def _rust_test_common(ctx):
```

Comment:
nit: Did this need to change? If `_rust_test_impl` is just gonna call `_rust_test_common`, a function that's only parameter is `ctx` then why the layer of abstraction?

Reply:
Before this PR, the output file was declared in `_rust_test_impl`. As per the changes, it now has a `test_{hash}` subdirectory, and in order to obtain the hash we need to have the crate root (or the symlink to it in case we had to create such a symlink in order to support generated files). So I would either have to move a lot of the logic from `_rust_test_common` to `_rust_test_impl` in order to be able to create the output filename, or to move the declaration of the output file into `_rust_test_common`. I went with the second option, but I agree that the layer of abstraction is not necessary anymore, and as such I removed `_rust_test_common` altogether now.
The reason why I put the test outputs into a `test_{hash}` directory instead of just `test` or something similar is to avoid conflicts with source directories.

I'll go ahead and merge this PR as you've marked this comment as a nit, but feel free to continue the discussion.